packages/hardhat-core/src/internal/core/providers/construction.ts

```diff
@@ -72,7 +72,9 @@ export function createProvider(
     ) {
       forkConfig = {
         jsonRpcUrl: hardhatNetConfig.forking?.url,
-        blockNumber: hardhatNetConfig.forking?.blockNumber,
+        blockNumber: BigIntUtils.mapBigIntToNumber(
```

Comment:
I think this function is used exclusively with `| undefined`. Maybe we can make it more specific?

Reply:
(we can ignore this comment)